### ISSUE TITLE
Add a single setting to scale all font sizes

### ISSUE THREAD

**Issue by joelostblom** [vega/vega#2946]
This was originally proposed for vegalite and altair in https://github.com/vega/vega-lite/issues/1714 and https://github.com/vega/vega-lite/issues/6988 and reccomended by @domoritz to open an issue here instead.

Currently, it is cumbersome to scale ip all font sizes since it involves setting many different parameters individually, for example:

```
spec.width = 800;
spec.height = 500;
spec.config.axisY.titleFontSize = 14;
spec.config.axisX.titleFontSize = 14;
spec.config.axisY.labelFontSize = 16;
spec.config.axisX.labelFontSize = 16;
spec.title.fontSize = 20;
spec.encoding.color.legend.titleFontSize = 15;
spec.encoding.color.legend.labelFontSize = 15;
```

Providing a shortcut for scaling all these parameters would be highly convenient from a usability perspective, particularly as a setting that applies to all plots (similar to setting a default theme), but also (less importantly) for individual plots. For what it's worth, this setting is available in other high level plotting libraries such as seaborn (`sns.set_theme(font_scale=1.5)`) and ggplot (`theme_set(theme_grey(base_size = 18))`).

I believe this feature is especially important since the the default font sizes in Vega are quite small. If possible, it would also be great to also consider an increase of the default fontsizes to ~1.2 times the current size, but I understand if this default is no longer amendable to change from suggestions.

Please:

- [x] Check for duplicate issues. Please file separate requests as separate issues on GitHub.
- [x] Clearly describe the goal of the feature request. Why is it needed?
- [x] Describe an envisioned solution. What do you want the new behavior to be?
- [ ] If applicable, share mockup images or links to examples illustrating the desired output.
- [ ] If applicable, share JSON spec designs for how the behavior would be made available in Vega.

**Comment by kanitw** [comment]
FYI, for 

```
spec.config.axisY.titleFontSize = 14;
spec.config.axisX.titleFontSize = 14;
spec.config.axisY.labelFontSize = 16;
spec.config.axisX.labelFontSize = 16;
```

you can just do 

```
spec.config.axis.titleFontSize = 14;
spec.config.axis.labelFontSize = 16;
```

So it's a bit less tedious.

**Comment by dlukes** [comment]
This would be a very useful feature indeed! One small nitpick: should it really only apply to font sizes, as the issue title says? The issue description itself mentions `spec.width` and `spec.height`, and mark sizes should probably scale along as well, shouldn't they?

**Comment by joelostblom** [comment]
Eventually it would be great if there was one setting for fonts and one setting for other chart elements. Similar to how it works in e.g. seaborn. For now, I think the most useful next feature would be scaling of font sizes as the default font size is rather small and makes titles etc, less legible than other chart elements such as marks etc.

**Comment by dlukes** [comment]
> the default font size is rather small and makes titles etc, less legible than other chart elements such as marks etc.

That's true, but which one of those would then affect the scaling of the overall figure size (`spec.width` and `spec.height` in your OP)? Or would those need to be set separately as well?

Since you brought up `seaborn` as prior art, how is this handled there? I confess I always just increase `figure.dpi` with `matplotlib`-based plotting libraries when I need to make a figure more readable, that's the option that I think of as "make everything bigger without worrying about the details", even though it's technically about pixel density.

**Comment by joelostblom** [comment]
In Seaborn's [set_theme](http://seaborn.pydata.org/generated/seaborn.set_theme.html#seaborn.set_theme) function, you can change the scaling of all plot elements (so text, axes, markers, etc) via the `plotting_context` param, and control the size of just the text with the `font_scale` param. For ggplot I am unsure if there is an option to control the size of all plot elements, but `base_size` can be used to scale all text as above.
